### DISCUSSION TITLE
Task supersedence

### DISCUSSION THREAD

**Discussion by aaronadamsCA** [vercel/turborepo#9037] [Ideas]
### Goals

Increase the frequency of "full turbo" checks.

### Background

It would be nice if I could tell the task graph that a given task can/should have its results superseded by those of another task.

For example, I could tell the graph that a `lint` result can be superseded by a `lint:fix` result, so that the `lint` task could be skipped if the `lint:fix` task already passed.

In a large and/or deep task graph this could offer a nice speedup.

### Proposal

I'm not sure how this would be implemented, I wasn't even sure what to call it.

I doubt I'm the first person with this idea, but I couldn't find any relevant discussions.

**Comment by anthonyshew** [comment]
Making sure that I've wrapped my head around this properly, you want to express "I already ran Task A in Package Z, so I want Task B in Package Z to get skipped, because I know it's going to pass." Do I have that right?
